### PR TITLE
Add basic consul scenario

### PR DESCRIPTION
**Pull request by Sgitario**
Module that covers the integration with Consul and ensure that changes in consul will take effect in the application.
Concretely, this module implements a resource that returns `Hello {some.property}`. By default, `some.property` is `Default`. If we change this property in consul and reload the application, we verify that the new property is taken from consul.

### CODE REVIEW

**Review by mjurc — APPROVED** [review]
LGTM.

**Comment by rsvoboda** [comment]
@kshpak, please take a look, waits for your review now

**Review by kshpak — CHANGES_REQUESTED** [review]
It's failing for me:
```
reetingResourceTest.java:[62,12] cannot find symbol
  symbol:   method start()
  location: variable app of type io.quarkus.test.QuarkusProdModeTest
```
the same with method stop()

**Comment by Sgitario** [comment]
> It's failing for me:
> 
> ```
> reetingResourceTest.java:[62,12] cannot find symbol
>   symbol:   method start()
>   location: variable app of type io.quarkus.test.QuarkusProdModeTest
> ```
> 
> the same with method stop()

You need to run this module with the latest changes in Quarkus upstream. (Make sure you have fetched the changes from upstream into your fork).

**Comment by kshpak** [comment]
@Sgitario I simply forget to fetch, you're right. Didn't get my coffee yet :D sry

**Review by kshpak — APPROVED** [review]

**Comment by rsvoboda** [comment]
@kshpak please merge, you should have right for it

File: 019-quarkus-consul/src/test/java/io/quarkus/qe/containers/ConsulTestResource.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    private GenericContainer<?> resource;
+
+    @SuppressWarnings("resource")
+    @Override
+    public Map<String, String> start() {
+
+        resource = new GenericContainer<>("consul:1.7")
+                .waitingFor(new LogMessageWaitStrategy().withRegEx(".*Node info in sync.*\\s"))
+                .withStartupTimeout(Duration.ofMillis(20000))
+                .withNetworkMode("host")
```
Comment (rsvoboda) [inline 019-quarkus-consul/src/test/java/io/quarkus/qe/containers/ConsulTestResource.java:31]: This won't work on Windows a macOS!

**Pull request merged**